Lux/src/lux_web.cpp

```diff
@@ -797,8 +800,567 @@ std::string get_recording_state() {
     }
 }
 
+// Camera-specific functions for live processing
+bool add_camera_frame(val image_data, int width, int height, std::string temp_name = "live_camera") {
+    try {
+        // Create a temporary image for camera frame
+        auto temp_img_ptr = std::make_unique<uimage>(vec2i{width, height});
+        if (!temp_img_ptr) {
+            std::cerr << "Failed to create camera frame image" << std::endl;
+            return false;
+        }
+
+        uimage temp_img = *temp_img_ptr;
+        unsigned char* img_buffer = (unsigned char*)temp_img.get_base_ptr();
+        size_t buffer_length = width * height * 4;
+
+        // Copy the data from the TypedArray (camera frame)
+        for (unsigned int i = 0; i < buffer_length; ++i) {
+            img_buffer[i] = image_data[i].as<uint8_t>();
+        }
+
+        // Add to scene buffers as temporary camera source
+        ubuf_ptr camera_buf(new buffer_pair<ucolor>(temp_img));
```

Comment:
Does this allocate a new buffer every frame? Best to keep a buffer active in memory and just reuse it - this is how the frame buffer in global context is used. I see UltraCameraBuf below - is this related